code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -787,41 +834,82 @@
 	description = "Efficiently restores brain damage."
 	color = "#A0A0A0" //mannitol is light grey, neurine is lighter grey
 	ph = 10.4
+	overdose_threshold = 15
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+	purity = REAGENT_STANDARD_PUIRTY
+	impure_chem = /datum/reagent/impurity/mannitol
+	inverse_chem_val = 0.45
+	impure_chem = /datum/reagent/impurity/mannitol
 
-/datum/reagent/medicine/mannitol/on_mob_life(mob/living/carbon/C, delta_time, times_fired)
-	C.adjustOrganLoss(ORGAN_SLOT_BRAIN, -2 * REM * delta_time)
+/datum/reagent/medicine/mannitol/on_mob_life(mob/living/carbon/owner, delta_time, times_fired)
+	owner.adjustOrganLoss(ORGAN_SLOT_BRAIN, -2 * REM * delta_time * normalise_creation_purity())
 	..()
 
 //Having mannitol in you will pause the brain damage from brain tumor (so it heals an even 2 brain damage instead of 1.8)
-/datum/reagent/medicine/mannitol/on_mob_metabolize(mob/living/carbon/C)
+/datum/reagent/medicine/mannitol/on_mob_metabolize(mob/living/carbon/owner)
 	. = ..()
-	ADD_TRAIT(C, TRAIT_TUMOR_SUPPRESSED, TRAIT_GENERIC)
+	ADD_TRAIT(owner, TRAIT_TUMOR_SUPPRESSED, TRAIT_GENERIC)
 
-/datum/reagent/medicine/mannitol/on_mob_end_metabolize(mob/living/carbon/C)
-	REMOVE_TRAIT(C, TRAIT_TUMOR_SUPPRESSED, TRAIT_GENERIC)
+/datum/reagent/medicine/mannitol/on_mob_end_metabolize(mob/living/carbon/owner)
+	REMOVE_TRAIT(owner, TRAIT_TUMOR_SUPPRESSED, TRAIT_GENERIC)
 	. = ..()
 
+/datum/reagent/medicine/mannitol/overdose_start(mob/living/owner)
+	to_chat(owner, "<span class='notice'>You suddenly feel </span><span class='purple'>E N L I G H T E N E D!</span>")
+
+/datum/reagent/medicine/mannitol/overdose_process(mob/living/owner)
```

Comment:
this should be changed to use delta_time

Reply:
Done!